src/comments.js

```diff
@@ -520,13 +520,20 @@ function handleTemplateLiteralComments(enclosingNode, comment) {
       comment
     );
     // Enforce all comments to be leading block comments.
-    comment.type = "CommentBlock";
+    makeBlockComment(comment);
     addTrailingComment(enclosingNode.expressions[expressionIndex], comment);
     return true;
   }
   return false;
 }
 
+function makeBlockComment(comment) {
+  if (!util.isBlockComment(comment)) {
+    comment.value += " ";
+  }
+  comment.type = "CommentBlock";
```

Comment:
Would you mind taking a look at merging the implementation of comments inside of jsx and comments inside of template literals?

The handling of comments inside of jsx is working really well and doesn't do any of those things. I think it would be better to unify the two than to keep piling on hacks.